### ISSUE TITLE
[HOLD for payment 2025-02-05] [$250] Expense shared with accountant shows incorrect rate 

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#46897]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.17-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @paultsimura 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1722892285723489

## Action Performed:
1. Open the chat with yourself
2. Submit a Distance request
3. In the Concierge prompt, click "Share it with my accountant"
4. Select a workspace and a different rate
5. Submit the request
6. Navigate to the request details page

## Expected Result:

- The Rate field shows the value selected while sharing the expense.
- There should be a proper descriptive "shared the expense" report action in the thread or no action at all 

## Actual Result:

- The Rate field shows a default p2p rate ($0.89)
- There is a "changed the expense" report action in the thread

## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/b161ebd9-eefb-46aa-97c7-f77a8fd373c5

https://github.com/user-attachments/assets/3f69d06a-b047-4f72-b5e4-1c99275ba94c

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019b051306a76e7ed6</li>
        <li>Upwork Job ID: 1821591099314477155</li>
        <li>Last Price Increase: 2024-10-24</li>
<li>Automatic offers: </li>
<ul>
<li>paultsimura | Contributor | 104594407</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @Pujan92 </details>

<details><summary>Issue Owner</summary>Current Issue Owner: @sonialiap / @mallenexpensify</details>

**Comment by wildan-m** [comment]
How can we enable the rate option for personal track expense? I think distance rate feature only available in workspace

**Comment by sonialiap** [comment]
@wildan-m I believe you're correct that you can't change the rate on a personal expense, but you don't need to do that. For this issue you want to have a workspace with multiple rates. Then, once you share the expense with someone on that workspace you will have the option to select one of the workspace rates

**Comment by sonialiap** [comment]
I'm OOO Aug 19-30, adding leave buddy

Status: Status: waiting for/ c+ reviewing proposals

**Comment by mallenexpensify** [comment]
Thanks @paultsimura , put on hold, removed `Help Wanted`

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989

**Comment by mallenexpensify** [comment]
- https://github.com/Expensify/App/pull/47136#issuecomment-2353603707

The above hit production 4 days ago, threw `retest weekly` on this

**Comment by mvtglobally** [comment]
Issue is reproducible during KI retests. 

https://github.com/user-attachments/assets/092bbfae-053f-4ca7-a006-eada2b1283a6

**Comment by mallenexpensify** [comment]
@paultsimura any thougths/ideas/tips here since the issue we held on didn't fix the bug? I see this from two months ago. 
> I’m planning to address a couple of Rate-related issues in it holistically. Currently we are discussing potential BE changes that will help eliminate multiple existing bugs easier.

I added back `Help Wanted` to get more 👀. Added to #expense project too.

**Comment by mallenexpensify** [comment]
@Pujan92 👀 on @paultsimura 's proposal above. thx

**Comment by Pujan92** [comment]
@paultsimura's [proposal](https://github.com/Expensify/App/issues/46897#issuecomment-2418927514) makes sense and looks good to me.

🎀👀🎀 C+ reviewed

**Comment by neil-marcellini** [comment]
I have some questions for you @paultsimura about your proposal. I'm going to ask them [in Slack here](https://expensify.slack.com/archives/C01GTK53T8Q/p1729611433691999) so we can hopefully have a quick conversation.

**Comment by neil-marcellini** [comment]
I'lll try to start on the backend for this today or tomorrow

**Comment by neil-marcellini** [comment]
I made a few changes on the backend and wrote a failing test for this. There's [some discussion](https://expensify.slack.com/archives/C01GTK53T8Q/p1730408521412489?thread_ts=1729611433.691999&cid=C01GTK53T8Q) about whether we should disable editing during the share flow or not. I don't think we should disable it, but I'll focus on getting the rate working and I hope to get some feedback from others before starting on making the distance editing work too.

**Comment by neil-marcellini** [comment]
The recent waypoints test is passing now. I also verified the distance and receipt filename. I need to finish verifying the receipt details and maybe push the Onyx update for the transaction. Might be worth manual testing first to see if that's needed.

**Comment by neil-marcellini** [comment]
The Auth tests I added are passing. Looks like there's some existing tests and other checks to fix on that PR. After that I can manually test. @paultsimura please try to move the App PR forward, and again lmk if you want me to make the backend available. It's a bit tricky since I can't work on anything else that requires me changing branches, so we'll have to see if the timing lines up. I might make some commits on that PR when I manually test.

**Comment by mallenexpensify** [comment]
@sonialiap I'm off the next week, can you please keep an eye on this issue til I'm back? I don't think any actions will be needed.

**Comment by neil-marcellini** [comment]
@NickTooker may we please have a copy check based on [this comment](https://github.com/Expensify/App/pull/51517#discussion_r1817699898)?

**Comment by NickTooker** [comment]
@neil-marcellini and I discussed the copy in Slack. We've agreed on the final version of the below:

`Rate not valid for this workspace. Please select an available rate from the workspace.`

`moved expense from self DM to chat with [name]`

**Comment by NickTooker** [comment]
Add me back to this issue if/when I am needed please.

**Comment by mallenexpensify** [comment]
> moved expense from self DM to chat with [name]

Does the 'm' in moved above need to be capitalized?

**Comment by mallenexpensify** [comment]
hmm... wonder why @paultsimura is [mentioned twice above](https://github.com/Expensify/App/issues/46897#issuecomment-2541736005). Going to :donothing: since I've yet to see this before

<img width="521" alt="SnagitHelper2024 2024-12-13 16 41 00" src="https://github.com/user-attachments/assets/df09286e-c686-4dff-9867-39c72806a8c4" />

**Comment by mallenexpensify** [comment]
I removed the `Awaiting Payment` label and `hold for payment` in the title since the PR was reverted.

**Comment by sonialiap** [comment]
Not overdue, the fix is being reworked

**Comment by sonialiap** [comment]
Neil is OOO through the end of this week. Looks like he should be back on Monday the 13th

**Comment by neil-marcellini** [comment]
I'm back and looks like I need to research the root cause of the [problem reported here](https://github.com/Expensify/App/pull/54669#issuecomment-2565944579). I'll work on it at a weekly pace since it's not very high priority

**Comment by neil-marcellini** [comment]
I couldn't reproduce said problem so it's back in the hands of the contributor.

**Comment by sonialiap** [comment]
Payment summary
- @paultsimura $250 - paid in upwork
- @Pujan92 $250 NewDot - please complete the [checklist](https://github.com/Expensify/App/issues/46897#issuecomment-2620812415)

**Comment by Pujan92** [comment]
# BugZero Checklist:

- [X] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [X] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production
  - [ ] 2b. Reported on staging (deploy blocker)
  - [X] 2c. Reported on both staging and production
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [X] 3c. Contributor
  - [] 3d. QA
  - [ ] 3z. Other:

</details>

- [X] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment: https://github.com/Expensify/App/pull/38543/files#r1944124101

- [X] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion: N/A

- [X] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal

Prerequisite: You are owner or a part of a workspace where custom distance rates are available.

1. Track the distance request 
2. Select the option "Share it with my accountant" for that request
3. Select a workspace and choose a different distance rate
4. Submit the request and navigate to the request details page
5. Verify the Rate value shows the correct distance rate and not the default rate

Do we agree 👍 or 👎

**Comment by garrettmknight** [comment]
$250 approved for @Pujan92
